controllers/components/dashboard/dashboard_controller_actions.go

```diff
@@ -0,0 +1,135 @@
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/annotations"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = []odhtypes.ManifestInfo{defaultManifestInfo(rr.Release.Name)}
+
+	extraParamsMap, err := computeKustomizeVariable(ctx, rr.Client, rr.Release.Name, &rr.DSCI.Spec)
+	if err != nil {
+		return errors.New("failed to set variable for extraParamsMap")
+	}
+
+	if err := odhdeploy.ApplyParams(rr.Manifests[0].String(), nil, extraParamsMap); err != nil {
+		return fmt.Errorf("failed to update params.env  from %s : %w", rr.Manifests[0].String(), err)
+	}
+
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	dashboard, ok := rr.Instance.(*componentsv1.Dashboard)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Dashboard)", rr.Instance)
+	}
+
+	if dashboard.Spec.DevFlags == nil {
+		return nil
+	}
+	// Implement devflags support logic
+	// If dev flags are set, update default manifests path
+	if len(dashboard.Spec.DevFlags.Manifests) != 0 {
+		manifestConfig := dashboard.Spec.DevFlags.Manifests[0]
+		if err := odhdeploy.DownloadManifests(ctx, ComponentNameUpstream, manifestConfig); err != nil {
+			return err
+		}
+		if manifestConfig.SourcePath != "" {
+			rr.Manifests[0].Path = odhdeploy.DefaultManifestPath
+			rr.Manifests[0].ContextDir = ComponentNameUpstream
+			rr.Manifests[0].SourcePath = manifestConfig.SourcePath
+		}
+	}
+
+	return nil
+}
+
+func customizeResources(_ context.Context, rr *odhtypes.ReconciliationRequest) error {
+	for i := range rr.Resources {
+		if rr.Resources[i].GroupVersionKind() == gvk.OdhDashboardConfig {
+			// mark the resource as not supposed to be managed by the operator
+			resources.SetAnnotation(&rr.Resources[i], annotations.ManagedByODHOperator, "false")
+			break
+		}
+	}
+
+	return nil
+}
+
+func updatePodSecurityRolebinding(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	switch rr.Release.Name {
+	case cluster.SelfManagedRhods, cluster.ManagedRhods:
+		if err := cluster.UpdatePodSecurityRolebinding(ctx, rr.Client, rr.DSCI.Spec.ApplicationsNamespace, "rhods-dashboard"); err != nil {
+			return fmt.Errorf("failed to update PodSecurityRolebinding for rhods-dashboard: %w", err)
+		}
+
+		err := rr.AddResource(&corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "anaconda-ce-access",
+				Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+			},
+			Type: corev1.SecretTypeOpaque,
+		})
+
+		if err != nil {
+			return fmt.Errorf("failed to create access-secret for anaconda: %w", err)
+		}
```

Comment:
I presume this is the approach that we already take in the existing code, but since I'm not really familiar with even that existing code, I'll ask here: why do we add this secret resource just for RHOAI and not for ODH?

Reply:
Yep this has not changed, I let @VaishnaviHire or @zdtsw to add additional context